manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -102,28 +103,35 @@ public Processor createProcessor(String bridgeId, String customerId, ProcessorRe
         }
 
         Processor newProcessor = new Processor();
+        newProcessor.setName(processorRequest.getName());
+        newProcessor.setSubmittedAt(ZonedDateTime.now());
+        newProcessor.setStatus(ManagedResourceStatus.ACCEPTED);
+        newProcessor.setBridge(bridge);
+        newProcessor.setShardId(shardService.getAssignedShardId(newProcessor.getId()));
 
         Set<BaseFilter> requestedFilters = processorRequest.getFilters();
 
         String requestedTransformationTemplate = processorRequest.getTransformationTemplate();
-        BaseAction requestedAction = processorRequest.getAction();
 
-        BaseAction resolvedAction = actionConfigurator.getResolver(requestedAction.getType())
-                .map(resolver -> resolver.resolve(requestedAction, customerId, bridge.getId(), newProcessor.getId()))
-                .orElse(requestedAction);
+        boolean isSourceProcessor = processorRequest.getSource() != null;
 
-        newProcessor.setName(processorRequest.getName());
-        newProcessor.setSubmittedAt(ZonedDateTime.now());
-        newProcessor.setStatus(ManagedResourceStatus.ACCEPTED);
-        newProcessor.setBridge(bridge);
-        newProcessor.setShardId(shardService.getAssignedShardId(newProcessor.getId()));
+        Action resolvedAction = isSourceProcessor
+                ? resolveSource(processorRequest.getSource(), customerId, bridge.getId(), newProcessor.getId())
+                : resolveAction(processorRequest.getAction(), customerId, bridge.getId(), newProcessor.getId());
+
+        ProcessorDefinition definition = isSourceProcessor
+                ? new ProcessorDefinition(requestedFilters, requestedTransformationTemplate, processorRequest.getSource(), resolvedAction)
+                : new ProcessorDefinition(requestedFilters, requestedTransformationTemplate, processorRequest.getAction(), resolvedAction);
 
-        ProcessorDefinition definition = new ProcessorDefinition(requestedFilters, requestedTransformationTemplate, requestedAction, resolvedAction);
         newProcessor.setDefinition(definitionToJsonNode(definition));
 
         // Processor, Connector and Work should always be created in the same transaction
         processorDAO.persist(newProcessor);
-        connectorService.createConnectorEntity(newProcessor, requestedAction);
+        if (isSourceProcessor) {
```

Comment:
I think with the `getType()` on the `Processor` we don't need to change the interface of the `ConnectorService`

Reply:
I think too :+1: